wazo_dird/plugins/office365_backend/services.py

```diff
@@ -24,105 +24,58 @@ class Office365Service(SelfSortingServiceMixin):
     USER_AGENT = 'wazo_ua/1.0'
 
     def get_contacts(self, microsoft_token, url, **list_params):
-        count, has_multiple_pages = self._get_total_contacts(microsoft_token, url)
-        contacts = list(self._fetch(microsoft_token, url, count, has_multiple_pages))
+        count = self._get_total_contacts(microsoft_token, url)
+        contacts = list(self._fetch(microsoft_token, url, count))
         total_contacts = len(contacts)
         sorted_contacts = self.sort(contacts, **list_params)
         paginated_contacts = self._paginate(sorted_contacts, **list_params)
         return paginated_contacts, total_contacts
```

Comment:
The `list_params` could be used for the microsoft API query in order to avoid over-querying for better performance. For example, listing the first 30 contacts would translate to `self._fetch(microsoft_token, url, list_params['limit'])`. The microsoft API seems to support sorting and filtering based on the contact attributes: https://learn.microsoft.com/en-us/previous-versions/office/office-365-api/api/version-2.0/complex-types-for-mail-contacts-calendar#OdataQueryParams.

Reply:
Thank you for your feedback.

Be careful, you pointed to the deprecated Outlook API.

There is a quota for the Microsoft Graph API, based on the number of requests. This is why I request all the contacts. Less requests => quota OK.

I'm not sure requesting 1K or 5K contacts from Microsoft will change their request time.

If we use `list_params['limit']` we must "store" somewhere the `@odata.nextLink` (there is no more `skip` parameter)
https://learn.microsoft.com/en-us/graph/paging?context=graph%2Fapi%2F1.0&view=graph-rest-1.0
https://learn.microsoft.com/en-us/graph/api/user-list-contacts?view=graph-rest-1.0&tabs=http